engine/flink/test-util/src/main/scala/pl/touk/nussknacker/engine/flink/test/ClassExtractionBaseTest.scala

```diff
@@ -0,0 +1,70 @@
+package pl.touk.nussknacker.engine.flink.test
+
+import io.circe.generic.semiauto.deriveDecoder
+import io.circe.parser.parse
+import io.circe.syntax.EncoderOps
+import io.circe.{Decoder, Json}
+import org.apache.commons.io.IOUtils
+import org.scalatest.{FunSuite, Matchers}
+import org.springframework.util.ClassUtils
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.typed.TypingResultDecoder
+import pl.touk.nussknacker.engine.api.typed.typing.TypedClass
+import pl.touk.nussknacker.engine.definition.ProcessDefinitionExtractor
+import pl.touk.nussknacker.engine.definition.TypeInfos.{ClazzDefinition, MethodInfo, Parameter}
+
+trait ClassExtractionBaseTest extends FunSuite with Matchers {
+
+  protected def model: ModelData
+
+  protected def outputResource: String
+
+  test("check extracted class for model") {
+    val types = ProcessDefinitionExtractor.extractTypes(model.processWithObjectsDefinition)
+    val parsed =  parse(IOUtils.toString(getClass.getResourceAsStream(outputResource))).right.get
+    val decoded = decode(parsed)
+
+    //println(types.asJson.spaces2)
+    checkGeneratedClasses(types, decoded)
+  }
+
+  //We don't do 'types shouldBe decoded' to avoid unreadable messages
+  private def checkGeneratedClasses(types: Set[ClazzDefinition], decoded: Set[ClazzDefinition]): Unit = {
+    types.size shouldBe decoded.size
```

Comment:
IMO we shouldn't fail so fast. Consider case when one new class is added. It would be nice to see full details of added class. Maybe in case of some differences would be better to always print:
- recently added classes
- recently removed classes
- recently modified methods inside each modified class
WDYT?

Reply:
:+1: , for methods I think 
`methods.keys shouldBe decodedMethods.keys` is sufficient